bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/member/RequestMembershipFlow.kt

```diff
@@ -65,12 +66,16 @@ open class RequestMembershipFlow(bno : Party, private val membershipMetadata : A
         }
         progressTracker.currentStep = ACCEPTING_INCOMING_PENDING_MEMBERSHIP
 
-        val selfSignedTx = subFlow(signResponder)
+        if (ourIdentity != bno) {
+            val selfSignedTx = subFlow(signResponder)
 
-        return if (bnoSession.getCounterpartyFlowInfo().flowVersion == 1) {
-            selfSignedTx
+            return if (bnoSession.getCounterpartyFlowInfo().flowVersion != 1) {
+                subFlow(ReceiveFinalityFlow(bnoSession, selfSignedTx.id))
+            } else {
+                selfSignedTx
+            }
         } else {
-            subFlow(ReceiveFinalityFlow(bnoSession, selfSignedTx.id))
+            return bnoSession.receive<SignedTransaction>().unwrap { it }
```

Comment:
Can we return `selfSignedTx` here instead?

Reply:
We don't have the selfSignedTx anymore since we aren't using the CollectSignaturesFlow on the Responder flow and in turn aren't using the SignTransactionFlow in the case where the BNO is the requesting member.  Thus we have to send it back from the Responder flow manually.

Reply:
Makes sense.